src/btree/bt_sync_obsolete.c

```diff
@@ -216,8 +216,9 @@ __sync_obsolete_cleanup_one(WT_SESSION_IMPL *session, WT_REF *ref)
      * checks. That's OK - in the worst case we might not review the ref this time, but we will on
      * subsequent reconciliations.
      */
-    if (ref->state == WT_REF_DELETED || ref->state == WT_REF_DISK) {
-        WT_REF_LOCK(session, ref, &previous_state);
+    ref_state = __wt_ref_get_state(ref);
```

Comment:
If we take this change the comment should be updated - it is no longer true (I presume)